include/sound/sof/dai-intel.h

```diff
@@ -155,10 +162,13 @@ struct sof_ipc_dai_dmic_params {
 	uint16_t duty_min;	/**< Min. mic clock duty cycle in % (20..80) */
 	uint16_t duty_max;	/**< Max. mic clock duty cycle in % (min..80) */
 
+	uint32_t wake_up_time;     /**< Time from clock start to data (us) */
+	uint32_t mode_change_time; /**< Time from clock change to data (us) */
+
```

Comment:
The discussion on the alsa-mailing list is about a delay on stop? Is this a different concept here, what does "clock change to data" mean?
Also at the DMIC level the only thing you can do is really delay the time when data is used (on start), or keep the clock on while discarding data on stop.
What am I missing?

Reply:
The key issue there is that the microphone clock should not be made active and stop for a too short time. If the capture time has already exceeded the min. requirement there is no need to wait before stopping the clock: After 10s capture we can stop the clock immediately.

The mic data sheets do not contain any data for "clock delay on stop" so I don't prefer it as parameter. So I decided to take two relevant parameters those can be found from most microphones in the market and use them for the purpose. The mode_change_time can be set to zero if the microphone is single mode. There's also often a suspend time mentioned (clock rate is reduced to less than min. or stopped) but I think the FW driver can't utilize it. I'm not aware (yet at least) that a too short clock stop would cause some issue.

So, with this data we can also ensure that we don't stop clock too early after start (or mode change that's for future) .

I was planning to replace in the FW driver the hard coded 2 ms mute in the driver with approximation of wake_up_time. And also check in trigger stop if the clock can be stopped immediate or need to schedule a delayed stop task.